src/OpenSage.Game/Logic/GameLogic.cs

```diff
@@ -85,28 +85,28 @@ public GameObject CreateObject(ObjectDefinition objectDefinition, Player player)
         return gameObject;
     }
 
-    internal void OnObjectIdChanged(GameObject gameObject, uint oldObjectId)
+    internal void OnObjectIdChanged(GameObject gameObject, ObjectId oldObjectId)
     {
-        if (oldObjectId != 0)
+        if (oldObjectId.Id != 0)
         {
             SetObject(oldObjectId, null);
         }
 
-        SetObject(gameObject.ID, gameObject);
+        SetObject(gameObject.Id, gameObject);
     }
 
-    private void SetObject(uint objectId, GameObject gameObject)
+    private void SetObject(ObjectId objectId, GameObject gameObject)
     {
-        while (_objects.Count <= objectId)
+        while (_objects.Count <= objectId.Id)
         {
             _objects.Add(null);
         }
-        _objects[(int)objectId] = gameObject;
+        _objects[(int)objectId.Id] = gameObject;
     }
 
-    public GameObject GetObjectById(uint id)
+    public GameObject GetObjectById(ObjectId id)
     {
-        return _objects[(int)id];
+        return _objects[(int)id.Id];
```

Comment:
Is it worth calling this like RawValue or something so we don't have `id.id` calls which might be confusing?

Reply:
Chnaging the name is a good idea. I'm not 100% sold on the name `RawValue` as that kind of implies there exists some sort of more refined or "safer" version of the value. For now I changed it to `Index`, as that what an object ID ultimately is (but 99.5% of code doesn't need to know that).